File: lib/feature/payment/domain/repository/order_repository.dart
import 'package:dartz/dartz.dart';
import '../../../../shared/model/response_model.dart';
import '../model/response/orders.dart';

abstract class OrderRepository {
  Future<Either<ResponseModel, Orders>> getDetailOrder({
    required String reference,
  });

  Future<Either<ResponseModel, ResponseModel>> createOrderPayment({
    required String paymentMethod,
    required String reference,
  });
}

File: src/features/payment/domain/repository/order_repository.ts
import { Either } from "@/shared/utils/utility/either";
import { ResponseModel } from "@/shared/domain/model/response_model";
import { OrdersResponse } from "../model/response/orders_response";
import { CreatePaymentRequest } from "../model/request/create_payment_request";

export interface OrderRepository {
  getDetailOrder(reference: string, token?: string, forceRefresh?: boolean): Promise<Either<ResponseModel, OrdersResponse>>;
  checkOrderStatus(reference: string, token?: string): Promise<Either<ResponseModel, any>>;
  createPayment(request: CreatePaymentRequest, token?: string): Promise<Either<ResponseModel, any>>;
}
